### PR TITLE
SDKS-2814: Impressions Counts recorder (imp dedup)

### PR DESCRIPTION
**Pull request by javrudsky**
# iOS SDK

## What did you accomplish?
- Added impressions count recorder, and worker
- Added tests for new components
- Fixed issue in LRU cache

### CODE REVIEW

File: Split/FetcherEngine/Recorder/ImpressionsCountRecorderWorker.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    func flush() {
+        var rowCount = 0
+        var failedCounts = [ImpressionsCountPerFeature]()
+        repeat {
+            let counts = countsStorage.pop(count: kPopCount)
+            rowCount = counts.count
+            if rowCount > 0 {
+                Logger.d("Sending impressions")
```
Comment (EmilianoSanchez) [inline Split/FetcherEngine/Recorder/ImpressionsCountRecorderWorker.swift:30]: I would use `"Sending impressions counts"` to distinguish from ImpressionsRecorderWorker

File: Split/FetcherEngine/Recorder/ImpressionsCountRecorder.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private let restClient: RestClientImpressionsCount
+
+    init(restClient: RestClientImpressionsCount) {
+        self.restClient = restClient
+    }
+
+    func execute(_ counts: ImpressionsCount) throws {
+
+        if !restClient.isSdkServerAvailable() {
+            Logger.d("Server is not reachable. Impressions sending will be delayed when host is reachable")
```
Comment (EmilianoSanchez) [inline Split/FetcherEngine/Recorder/ImpressionsCountRecorder.swift:26]: `Impressions` --> `Impression counts` ?

**Review by EmilianoSanchez — APPROVED** [review]
LGTM Javi. There are just some comments on log messages.

File: Split/FetcherEngine/Recorder/ImpressionsCountRecorderWorker.swift (hunk 1 above)
Comment (javrudsky) [reply Split/FetcherEngine/Recorder/ImpressionsCountRecorderWorker.swift:30]: Fixed Doc!

File: Split/FetcherEngine/Recorder/ImpressionsCountRecorder.swift (hunk 2 above)
Comment (javrudsky) [reply Split/FetcherEngine/Recorder/ImpressionsCountRecorder.swift:26]: Also Fixed Doc! Thanks

**Pull request merged**